libs/orders/src/lib/components/order-list-manager/order-list-manager.tsx

```diff
@@ -30,16 +41,49 @@ export const OrderListManager = ({ partyId }: OrderListManagerProps) => {
     scrolledToTop.current = event.top <= 0;
   };
 
+  const onFilterChanged = (event: FilterChangedEvent) => {
+    const updatedFilter = event.api.getFilterModel();
+    if (Object.keys(updatedFilter).length) {
+      setFilter(updatedFilter);
+    } else if (filter) {
+      setFilter(undefined);
+    }
+  };
+
+  const onSortChange = (event: SortChangedEvent) => {
+    const sort = event.columnApi
+      .getColumnState()
+      .sort((a, b) => (a.sortIndex || 0) - (b.sortIndex || 0))
+      .reduce((acc, col) => {
+        if (col.sort) {
+          const { colId, sort } = col;
+          acc.push({ colId, sort });
+        }
+        return acc;
+      }, [] as { colId: string; sort: string }[]);
+    setSort(sort.length > 0 ? sort : undefined);
+  };
+
   return (
-    <AsyncRenderer loading={loading} error={error} data={data}>
+    <>
       <OrderList
         ref={gridRef}
-        rowModelType={data?.length ? 'infinite' : 'clientSide'}
-        rowData={data?.length ? undefined : []}
+        rowModelType="infinite"
         datasource={{ getRows }}
         onBodyScrollEnd={onBodyScrollEnd}
         onBodyScroll={onBodyScroll}
+        onFilterChanged={onFilterChanged}
+        onSortChanged={onSortChange}
       />
-    </AsyncRenderer>
+      <div className="pointer-events-none absolute inset-0 top-5">
```

Comment:
question (non blocking): This seems a bit weird. Why is OrderList not rendered within the AsyncRenderer?